ksso/main.py

```diff
@@ -79,46 +103,99 @@ def main():
             "where <name> is the actual client name/prefix"
         ),
     )
-    parser.add_argument(
+    login_parser.add_argument(
         "--aws-role-arn",
         type=str,
         required=True,
         help="AWS IAM Role ARN to assume (e.g., arn:aws:iam::123456789012:role/role-name)",
     )
+
+    # Logout command
+    logout_parser = subparsers.add_parser("logout", help="Logout and remove cached AWS credentials")
+    logout_parser.add_argument(
+        "--client-id",
+        type=str,
+        help="Keycloak client ID to remove credentials for (removes all if not specified)",
+    )
+    logout_parser.add_argument(
+        "--aws-role-arn",
+        type=str,
+        help="AWS IAM Role ARN to remove credentials for (requires --client-id)",
+    )
     args = parser.parse_args()
 
-    try:
-        ksso_config = load_config(args.config)
-    except ConfigError as e:
-        logger.error(e)
-        exit(1)
-    except (requests.RequestException, requests.exceptions.HTTPError, ValueError) as e:
-        logger.error(f"Error while connecting to Keycloak: {e}")
-        exit(1)
-
-    # Store dynamic values in Flask app config
-    client_id, aws_role_arn = args.client_id, args.aws_role_arn
-    app.config["KEYCLOAK_URL"] = ksso_config.sso_token_service_url
-    app.config["REDIRECT_URI"] = ksso_config.sso_redirect_url
-    app.config["CLIENT_ID"] = client_id
-    app.config["AWS_ROLE_ARN"] = aws_role_arn
-
-    # Start Flask app in a separate thread
-    threading.Thread(target=lambda: app.run(port=ksso_config.sso_agent_port), daemon=True).start()
-
-    # Open the browser for authentication
-    webbrowser.open(f"http://localhost:{ksso_config.sso_agent_port}/")
-
-    # Wait for the token and session name to be available in the queue
-    access_token, session_name = token_queue.get()
-    # Assume AWS role using the obtained access token and session name
-    credentials = assume_aws_role_with_keycloak_token(access_token, aws_role_arn, session_name)
+    # Set logging level based on verbose flag
+    if args.verbose:
+        logging.getLogger().setLevel(logging.INFO)
+        logging.getLogger("ksso").setLevel(logging.INFO)
+        logger.info("Verbose logging enabled")
+
+    # Handle logout command
+    if args.command == "logout":
```

Comment:
I suggest to move this logic to `logout` command itself too. So our main just consist from args parsing and few required function calls

Reply:
Done, good suggestion